java-runtime/ftl-runtime/runtime/src/main/java/xyz/block/ftl/runtime/FTLHttpHandler.java

```diff
@@ -0,0 +1,245 @@
+package xyz.block.ftl.runtime;
+
+import java.io.ByteArrayOutputStream;
+import java.net.InetSocketAddress;
+import java.nio.channels.Channels;
+import java.nio.channels.WritableByteChannel;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+import jakarta.inject.Singleton;
+import jakarta.ws.rs.core.MediaType;
+
+import org.jboss.logging.Logger;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.ByteString;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.FileRegion;
+import io.netty.handler.codec.http.*;
+import io.netty.util.ReferenceCountUtil;
+import io.quarkus.netty.runtime.virtual.VirtualClientConnection;
+import io.quarkus.netty.runtime.virtual.VirtualResponseHandler;
+import io.quarkus.vertx.http.runtime.QuarkusHttpHeaders;
+import io.quarkus.vertx.http.runtime.VertxHttpRecorder;
+import xyz.block.ftl.v1.CallRequest;
+import xyz.block.ftl.v1.CallResponse;
+
+@SuppressWarnings("unused")
+@Singleton
+public class FTLHttpHandler implements VerbInvoker {
```

Comment:
it looks like http ingress verbs will be registered to use this handler on invocation, is that right?

why does java/kotlin need its own ingress logic? shouldn't we be able to invoke these verbs as normal on the runtime side, since the FTL controller will be responsible for serving the HTTP endpoint and routing the inbound ingress requests appropriately?

Reply:
This is purely about the Java API that is used (i.e. allowing you to use JAX-RS instead of a custom HTTP annotations). From the invocations side everything works as normal.

Reply:
ahh got it